src/Compilers/Core/Portable/CodeGen/PrivateImplementationDetails.cs

```diff
@@ -110,8 +113,22 @@ internal void Freeze()
             }
 
             // Sort data fields
-            _orderedMappedFields = _mappedFields.Values.OrderBy((x, y) => x.Name.CompareTo(y.Name)).AsImmutable();
-            _orderedSynthesizedMethods = _synthesizedMethods.OrderBy(kvp => kvp.Key).Select(kvp => kvp.Value).AsImmutable();
+            ArrayBuilder<SynthesizedStaticField> fieldsBuilder = ArrayBuilder<SynthesizedStaticField>.GetInstance(_mappedFields.Count);
+            fieldsBuilder.AddRange(_mappedFields.Values.OrderBy((x, y) => x.Name.CompareTo(y.Name)));
+            if (_mvidField != null)
+            {
+                fieldsBuilder.Add(_mvidField);
```

Comment:
I doubt it will cause a problem - it just seems weird that this one field isn't sorted with the others.